back/engines/commercial/analysis/app/controllers/analysis/web_api/v1/analyses_controller.rb

```diff
@@ -67,12 +69,20 @@ def set_analysis
         end
 
         def analysis_params
-          params.require(:analysis).permit(:project_id, :phase_id, custom_fields_ids: [])
+          params.require(:analysis).permit(:project_id, :phase_id, custom_field_ids: [])
         end
 
         def side_fx_service
           @side_fx_service ||= SideFxAnalysisService.new
         end
+
+        def detect_custom_fields
+          container = @analysis.phase || @analysis.project
+          custom_form = container.custom_form
+          custom_fields = IdeaCustomFieldsService.new(custom_form).all_fields
+          # we map, because custom fields can be an array or a scope
+          custom_fields.map(&:id)
```

Comment:
@kogre  I'm not sure if this will successfully store the relationship between analysis and custom field. The issue is that the IDs of the default fields (which are not persisted) are randomly generated at runtime. So each time the default fields are requested, they have different IDs. Even if they would always have the same IDs (using the custom form ID as some sort of seed?), because the fields are not in the custom_fields table, I think you'll still get a DB error on the foreign key reference.

This seems to be a complicated problem. I don't like using JSON-type attributes (because of maintenance and data consistency issues), but perhaps storing an array of custom field keys/codes could be the easiest solution here?

Reply:
Well spotted, that's a good point.

An easier and more robust solution could generate these fields at the moment this is requested, do you see any issues with that approach?

Reply:
@kogre Yes, if that's an option then I think that's a good solution :)

Reply:
@sebastienhoorens after some more investigating it seems to me like this situation doesn't really happen.

We have [this](https://github.com/CitizenLabDotCo/citizenlab/blob/f4f547df00bf4f219b3162130e18c3d967bff424/back/app/services/side_fx_project_service.rb#L16) and [this](https://github.com/CitizenLabDotCo/citizenlab/blob/f4f547df00bf4f219b3162130e18c3d967bff424/back/app/services/side_fx_phase_service.rb#L16) in the project/phase side fx, which creates the form at creation time.

Any reason not to trust it?

Reply:
@kogre This one is only for native surveys and does not seem to create the default form for ideation.

Reply:
I never wished more that you were wrong @sebastienhoorens 
But of course you're not :) Digging a bit deeper then.